connector-servicenow/src/main/java/com/redhat/cloud/notifications/connector/servicenow/ServiceNowCloudEventDataExtractor.java

```diff
@@ -0,0 +1,56 @@
+package com.redhat.cloud.notifications.connector.servicenow;
+
+import com.redhat.cloud.notifications.connector.CloudEventDataExtractor;
+import io.vertx.core.json.JsonObject;
+import org.apache.camel.Exchange;
+import org.apache.commons.validator.routines.UrlValidator;
+import org.apache.http.ProtocolException;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import static com.redhat.cloud.notifications.connector.ExchangeProperty.ORG_ID;
+import static com.redhat.cloud.notifications.connector.ExchangeProperty.TARGET_URL;
+import static com.redhat.cloud.notifications.connector.servicenow.ExchangeProperty.ACCOUNT_ID;
+import static com.redhat.cloud.notifications.connector.servicenow.ExchangeProperty.AUTHENTICATION_TOKEN;
+import static com.redhat.cloud.notifications.connector.servicenow.ExchangeProperty.KAFKA_PROCESSOR;
+import static com.redhat.cloud.notifications.connector.servicenow.ExchangeProperty.TARGET_URL_NO_SCHEME;
+import static com.redhat.cloud.notifications.connector.servicenow.ExchangeProperty.TRUST_ALL;
+import static org.apache.commons.validator.routines.UrlValidator.ALLOW_LOCAL_URLS;
+
+@ApplicationScoped
+public class ServiceNowCloudEventDataExtractor extends CloudEventDataExtractor {
+
+    public static final String NOTIF_METADATA = "notif-metadata";
+
+    private static final UrlValidator HTTP_URL_VALIDATOR = new UrlValidator(new String[] {"http"}, ALLOW_LOCAL_URLS);
+    private static final UrlValidator HTTPS_URL_VALIDATOR = new UrlValidator(new String[] {"https"}, ALLOW_LOCAL_URLS);
```

Comment:
I guess in the future we will have to disallow the local URLs for stage & prod?

Reply:
I'm not happy with the current URL validation logic and the fact that half of it is done in `notifications-backend` and the other half here.

The behavior we have in this PR comes from the eventing app but I think we'll need to revisit that later.